core/src/main/clojure/xtdb/metrics.clj

```diff
@@ -42,6 +43,9 @@
        (cond-> (:unit opts) (.baseUnit (str (:unit opts))))
        (.register reg))))
 
+(defn add-allocator-gauge [reg meter-name ^BufferAllocator allocator]
+  (add-gauge reg meter-name (fn [] (.getAllocatedMemory allocator)) {:unit "bytes"}))
```

Comment:
If there were other metrics we were interested in here could fairly easily replace "meter-name" with "allocator-name" or something similar and register multiple different gauges here. `AllocatedMemory` seemed the most obviously useful to us as we move forward with stability work.